test/spec/controllers/eventDetails.spec.js

```diff
@@ -27,212 +27,252 @@ describe('Controller: paymentModal', function () {
+      expect(scope.conference.registrationPages[0].blocks[2].tag).toEqual(
+        'EFORM',
+      );
 
-    expect(scope.conference.registrationPages[0].blocks[3].title).toEqual(
-      'Guardian Email',
-    );
+      expect(scope.conference.registrationPages[0].blocks[1].title).toEqual(
+        'Are you under 18?',
+      );
 
-    expect(scope.conference.registrationPages[0].blocks[3].tag).toEqual(
-      'EFORM',
-    );
+      expect(scope.conference.registrationPages[0].blocks[1].tag).toEqual(
+        'EFORM',
+      );
 
-    expect(scope.conference.registrationPages[0].blocks[2].title).toEqual(
-      'Guardian Name',
-    );
+      expect(scope.conference.registrationPages[0].blocks.length).toBe(4);
 
-    expect(scope.conference.registrationPages[0].blocks[2].tag).toEqual(
-      'EFORM',
-    );
+      scope.updateLiabilityQuestions();
+      $httpBackend.flush();
 
-    expect(scope.conference.registrationPages[0].blocks[1].title).toEqual(
-      'Are you under 18?',
-    );
+      expect(scope.conference.registrationPages[0].blocks[3].tag).toEqual(null);
 
-    expect(scope.conference.registrationPages[0].blocks[1].tag).toEqual(
-      'EFORM',
-    );
+      expect(scope.conference.registrationPages[0].blocks[2].tag).toEqual(null);
 
-    expect(scope.conference.registrationPages[0].blocks.length).toBe(4);
+      expect(scope.conference.registrationPages[0].blocks[2].tag).toEqual(null);
+    });
 
-    scope.updateLiabilityQuestions();
-    $httpBackend.flush();
+    it('saveEvent() should validate the conference', () => {
+      scope.saveEvent();
+      //Expect that there will be event types given
+      expect(scope.getEventTypes().length).toBeGreaterThan(0);
+      expect(scope.notify.message.toString()).toContain(
+        'Please enter which Event Type',
+      );
+    });
+  });
 
-    expect(scope.conference.registrationPages[0].blocks[3].tag).toEqual(null);
+  describe('Conference without type', function () {
+    beforeEach(
+      angular.mock.inject(function (
+        $rootScope,
+        $controller,
+        _$uibModal_,
+        _testData_,
+        _$httpBackend_,
+      ) {
+        testData = _testData_;
+        scope = $rootScope.$new();
+        $httpBackend = _$httpBackend_;
+
+        testData.conference.type = null;
+
+        $controller('eventDetailsCtrl', {
+          $scope: scope,
+          conference: testData.conference,
+          currencies: testData.currencies,
+          $uibModal: _$uibModal_,
+          permissions: {},
+        });
+        scope.ministries = testData.ministries;
+      }),
+    );
 
-    expect(scope.conference.registrationPages[0].blocks[2].tag).toEqual(null);
+    it('saveEvent() should validate the Ministry Purpose', () => {
+      scope.saveEvent();
 
-    expect(scope.conference.registrationPages[0].blocks[2].tag).toEqual(null);
+      expect(scope.notify.message.toString()).toContain(
+        'Please enter Ministry Purpose.',
+      );
+    });
```

Comment:
Also another test when it's not a Cru event, that it shouldn't show any errors about event type?